tests/integration/explain/default/with_count_test.go

```diff
@@ -76,68 +76,76 @@ func TestExplainQueryOneToManyWithACount(t *testing.T) {
 			},
 		},
 
-		// ----> selectTopNode                (explainable but no-attributes)
-		//    ----> countNode                 (explainable)
-		//        ----> selectNode            (explainable)
-		//             ----> typeIndexJoin    (explainable)
-		//                 ----> typeJoinMany (non-explainable)
-		//                     ----> scanNode (explainable)
-		Results: []dataMap{
+		ExpectedPatterns: []dataMap{
 			{
 				"explain": dataMap{
 					"selectTopNode": dataMap{
 						"countNode": dataMap{
-							"sources": []dataMap{
-								{
-									"filter":    nil,
-									"fieldName": "books",
-								},
-							},
 							"selectNode": dataMap{
-								"filter": nil,
-								"typeIndexJoin": dataMap{
-									"joinType": "typeJoinMany",
-									"rootName": "author",
-									"root": dataMap{
-										"scanNode": dataMap{
-											"filter":         nil,
-											"collectionID":   "3",
-											"collectionName": "Author",
-											"spans": []dataMap{
-												{
-													"start": "/3",
-													"end":   "/4",
-												},
-											},
-										},
-									},
-									"subTypeName": "books",
-									"subType": dataMap{
-										"selectTopNode": dataMap{
-											"selectNode": dataMap{
-												"filter": nil,
-												"scanNode": dataMap{
-													"filter":         nil,
-													"collectionID":   "2",
-													"collectionName": "Book",
-													"spans": []dataMap{
-														{
-															"start": "/2",
-															"end":   "/3",
-														},
-													},
-												},
-											},
-										},
-									},
-								},
+								"typeIndexJoin": normalTypeJoinPattern,
 							},
 						},
 					},
 				},
 			},
 		},
+
+		ExpectedTargets: []explainUtils.PlanNodeTargetCase{
+			{
+				TargetNodeName:    "countNode",
+				IncludeChildNodes: false,
+				ExpectedAttributes: dataMap{
+					"sources": []dataMap{
+						{
+							"filter":    nil,
+							"fieldName": "books",
+						},
+					},
+				},
+			},
+			{
+				TargetNodeName:    "typeIndexJoin",
+				IncludeChildNodes: false,
+				ExpectedAttributes: dataMap{
+					"joinType":    "typeJoinMany",
+					"rootName":    "author",
+					"subTypeName": "books",
+				},
+			},
+			{
+				TargetNodeName:    "scanNode", // inside of root
+				OccurancesToSkip:  0,
+				IncludeChildNodes: true, // should be leaf of it's branch, so will have no child nodes.
```

Comment:
question: I saw this on the other PR I approved but now I'm starting to scratch my head when looking at the comment. Is it normal that the field says true to include child nodes but the comment says "will have no child node"?

Reply:
`scanNode` is a special case as it should be the last node (leaf) in all these cases. The `IncludeChildNodes: true` is there to ensure that there are no nodes inside / under the `scanNode`. If some child nodes were ever introduced in the future, it would ensure to fail the test (as assumptions have changed) and should deliberately handle/change the test then based on the new change.

Reply:
Ok thanks for the explanation. I do feel like the name is a little misleading though. Out of scope here of course but maybe something we can revisit later?

Reply:
`IncludeChildNodes` includes child nodes if they are available. I thought it covered the purpose well. Open to suggestions if you have any?

Reply:
Oh! I get it now. It’s the comment on hat threw me off. I thought the comment was saying that we set it to true because there will be no child nodes. Not “although” it’s set to true, there will be no child nodes.